src/server/dataPlaneHelper.ts

```diff
@@ -41,27 +41,37 @@ export const processDataPlaneResponse = (httpRes: request.Response, body: any):
     if (httpRes) {
         if (httpRes.headers && httpRes.headers[DEVICE_STATUS_HEADER]) { // handles happy failure cases when error code is returned as a header
             return {
-                body: {body: JSON.parse(body)},
+                body: {body: parseBody(body)},
                 statusCode: parseInt(httpRes.headers[DEVICE_STATUS_HEADER] as string) // tslint:disable-line:radix
             };
         }
         else {
             if (httpRes.statusCode >= SUCCESS && httpRes.statusCode < MULTIPLE_CHOICES) {
                 return {
-                    body: {body: JSON.parse(body), headers: httpRes.headers},
+                    body: {body: parseBody(body), headers: httpRes.headers},
                     statusCode: httpRes.statusCode
                 };
             } else {
                 return {
-                    body: {body: JSON.parse(body)},
+                    body: {body: parseBody(body)},
                     statusCode: httpRes.statusCode
                 };
             }
         }
     }
     else {
         return {
-            body: {body: JSON.parse(body)}
+            body: {body: parseBody(body)}
         };
     }
 };
+
+// tslint:disable-next-line:no-any
+const parseBody = (body: any) => {
```

Comment:
wouldn't body be a string?

Reply:
removed this, and instead add a try catch for the entire block and send 500 with undefined body back when something unexpected happens